packages/core/src/FileUploader/FileList/FileList.js

```diff
@@ -1,20 +1,22 @@
 import React from "react";
 import PropTypes from "prop-types";
 import { withStyles } from "@material-ui/core";
-import { setUid } from "../..";
+import { setId, useUniqueId } from "../..";
 import File from "../File";
 import styles from "./styles";
 
 const FileList = ({ id, classes, list = [], removeFileButtonLabel, onFileRemoved }) => {
+  const elementId = useUniqueId(id, "hvfilelist");
+
   const hasFiles = list.length > 0;
   if (!hasFiles) return null;
 
   return (
-    <ul id={id} className={classes.list}>
+    <ul id={setId(id, "list")} className={classes.list}>
```

Comment:
Here I would be passing an `id` to `FileList`, but that passed `id` will never be visible

```tsx
<FileList id="idsSuckBtw" />
```

And then if I do something like `document.getElementById("idsSuckBtw")` I would get nothing, because it would get some internal suffix I don't know about.

Also, we're changing the current existing `id`.
This could be considered a breaking change, as some users might be using the `id` for tests. But regarding this I'd say its their own fault [for not using better selectors](https://testing-library.com/docs/queries/about#priority) 😛